src/app/metadata/features/metadata-browser/metadata-browser.component.ts

```diff
@@ -161,6 +199,55 @@ export class MetadataBrowserComponent implements OnInit {
     this.#updateFacets(facetKey, facetOption, newValue);
   }
 
+  /**
+   * Turns a FacetFilterSetting dictionary into a string for usage in the url f parameter
+   * This is the inverse operation to the #facetDataFromString function
+   * @param facetData the dictionary to turn into a string
+   * @returns the string representation
+   */
+  #facetDataToString(facetData: FacetFilterSetting): string {
+    console.log(facetData);
+    if (Object.keys(facetData).length === 0) {
+      return '';
+    }
+    let facetStrings = [];
+    for (const key in facetData) {
+      for (const index in facetData[key]) {
+        facetStrings.push(key + ':' + facetData[key][index]);
+      }
+    }
+    const ret = facetStrings.join(';');
+    console.log(ret);
+    return ret;
+  }
+
+  /**
+   * This function takes a string (as used in the f argument of the url) and turns it into a FacetFilterSetting dictionary.
+   * This is the inverse operation of #facetDataToString.
+   * @param input the string to parse
+   * @returns The FacetFilterSetting dictionary format
+   */
+  #facetDataFromString(input: string): FacetFilterSetting {
+    console.log(input);
+    if (input.indexOf(':') == -1) {
+      return {};
+    }
+    const filterStrings = input.split(';');
+    const ret: FacetFilterSetting = {};
+    for (const facet of filterStrings) {
+      if (facet.indexOf(':') !== -1) {
+        const [facetName, facetValue] = facet.split(':');
+        if (ret[facetName]) {
+          ret[facetName].push(facetValue);
+        } else {
+          ret[facetName] = [facetValue];
+        }
+      }
```

Comment:
```suggestion
      const [facetName, facetValue] = facet.split(':' ,2);
      if (facetValue !== undefined) {
        if (ret[facetName]) {
          ret[facetName].push(facetValue);
        } else {
          ret[facetName] = [facetValue];
        }
      }
```

the 2 makes sure it also works with values that contain colons